### PR TITLE
🏗✨ Adopt `prettier` for code formatting

### PR DESCRIPTION
**Pull request by DerekNonGeneric**
Regardless of the result of https://github.com/ampproject/amp.dev/pull/3346, there is a very good chance that this still needs to happen. I'm following in the footsteps of https://github.com/ampproject/amphtml/pull/21212. Activity on that discussion (#3346) should inform further development.

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{
+  "bracketSpacing": false,
+  "singleQuote": true,
+  "trailingComma": "es5",
+  "quoteProps": "preserve",
+  "overrides": [
+    {
+      "files": "OWNERS",
```
Comment (sebastianbenz) [inline .prettierrc]: We don't need this part

File: package.json [hunk 2]
```diff
@@ -112,6 +112,7 @@
     "del": "5.1.0",
     "eslint": "6.8.0",
     "eslint-config-google": "0.14.0",
+    "eslint-plugin-prettier": "^3.1.2",
```
Comment (sebastianbenz) [inline package.json]: please pin versions

File: package.json [hunk 3]
```diff
@@ -145,6 +146,7 @@
     "octonode": "0.9.5",
     "optimize-css-assets-webpack-plugin": "5.0.3",
     "preload-webpack-plugin": "3.0.0-beta.4",
+    "prettier": "^1.19.1",
```
Comment (sebastianbenz) [inline package.json]: please pin versions

**Comment by DerekNonGeneric** [comment]
@sebastianbenz, this one should be good to go. I removed rules duplicated by the fact that eslintrc rules cascade into subdirs. We'll need this merged to proceed with the Markdown linting.

**Comment by DerekNonGeneric** [comment]
Fixed a nit in commit message: reword `npm run lint:node` -> `npm run fix:node`

**Comment by sebastianbenz** [comment]
Merged against future and re-created here  https://github.com/ampproject/amp.dev/pull/3426

**Pull request closed without merging**